Fix: Deploying Port-less Container Image Fails

gijohn opened the pull request:
Fixes: https://jira.coreos.com/browse/ODC-2060

gijohn commented:
/kind bug
/cc @andrewballantyne @vikram-raj 
/assign @rohitkrai03 @christianvogt

gijohn commented:
/retest

debsmita1 commented on frontend/packages/dev-console/src/utils/shared-submit-utils.ts:
```diff
@@ -10,6 +10,12 @@ export const annotations = {
+const isDeployImageFormData = (
+  formData: DeployImageFormData | GitImportFormData,
+): formData is DeployImageFormData => {
+  return (formData as DeployImageFormData).isi !== undefined;
```
May be write this as:
```suggestion
  return !_.isEmpty((formData as DeployImageFormData).isi);

christianvogt replied:
or `'isi' in formData` or `formData.hasOwnProperty('isis')` ?

andrewballantyne commented:
LGTM - Tested it and it seems to have nicely solved the issue.

gijohn commented:
/test e2e-aws-console-olm

andrewballantyne commented:
/lgtm

christianvogt commented:
@gijohn please add a unit test

gijohn commented:
> @gijohn please add a unit test

@christianvogt I have added another validation. Previously the test was written for a wrong logic. I have corrected that too.

christianvogt commented:
/approve
/lgtm
